pinot-common/src/main/java/com/linkedin/pinot/common/utils/helix/HelixHelper.java

```diff
@@ -95,13 +96,17 @@ public Boolean call() {
           BaseDataAccessor<ZNRecord> baseDataAccessor = dataAccessor.getBaseDataAccessor();
           boolean success;
 
+          // If the ideal state is large enough, enable compression
+          if (MAX_PARTITION_COUNT_IN_UNCOMPRESSED_IDEAL_STATE < updatedIdealState.getPartitionSet().size()) {
+            updatedIdealState.getRecord().setBooleanField("enableCompression", true);
+          }
+
           try {
             success =
                 baseDataAccessor.set(path, updatedIdealState.getRecord(), idealState.getRecord().getVersion(),
                     AccessOption.PERSISTENT);
           } catch (Exception e) {
-            LOGGER.warn("Caught exception: {} while updating ideal state for resource {}, retrying.", resourceName,
-                e.getMessage());
+            LOGGER.warn("Caught exception while updating ideal state for resource {}, retrying.", resourceName, e);
```

Comment:
The idea was to add it to the log message if we had an exception. The log message should include whether we tried compression or not.